storm-core/src/clj/org/apache/storm/config.clj

```diff
@@ -53,7 +54,7 @@
   [freq]
   (let [freq (int freq)
         start (int 0)
-        r (java.util.Random.)
+        r (XORShiftRandom.)
```

Comment:
Here too the code can be called from multiple threads, but the worst that happens is we double up on sampling some items for metrics.  Ideally this would have a comment explaining this.